src/app/clusters/chime-server/chime-server.cpp

```diff
@@ -0,0 +1,185 @@
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/****************************************************************************'
+ * @file
+ * @brief Implementation for the Chime Server Cluster
+ ***************************************************************************/
+
+#include "chime-server.h"
+
+#include <app/AttributeAccessInterfaceRegistry.h>
+#include <app/CommandHandlerInterfaceRegistry.h>
+#include <app/ConcreteAttributePath.h>
+#include <app/InteractionModelEngine.h>
+#include <app/util/attribute-storage.h>
+#include <app/util/util.h>
+#include <protocols/interaction_model/StatusCode.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::DataModel;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::Chime;
+using namespace chip::app::Clusters::Chime::Attributes;
+using chip::Protocols::InteractionModel::Status;
+
+namespace chip {
+namespace app {
+namespace Clusters {
+
+ChimeServer::ChimeServer(EndpointId endpointId, ChimeDelegate & delegate) :
+    AttributeAccessInterface(MakeOptional(endpointId), Chime::Id), CommandHandlerInterface(MakeOptional(endpointId), Chime::Id),
+    mDelegate(delegate)
+{
+    mDelegate.SetChimeServer(this);
+}
+
+ChimeServer::~ChimeServer()
+{
+    AttributeAccessInterfaceRegistry::Instance().Unregister(this);
+    CommandHandlerInterfaceRegistry::Instance().UnregisterCommandHandler(this);
+}
+
+CHIP_ERROR ChimeServer::Init()
+{
+    VerifyOrReturnError(AttributeAccessInterfaceRegistry::Instance().Register(this), CHIP_ERROR_INTERNAL);
+    ReturnErrorOnFailure(CommandHandlerInterfaceRegistry::Instance().RegisterCommandHandler(this));
+    return CHIP_NO_ERROR;
+}
+
+// AttributeAccessInterface
+CHIP_ERROR ChimeServer::Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder)
+{
+    VerifyOrDie(aPath.mClusterId == Chime::Id);
+
+    switch (aPath.mAttributeId)
+    {
+    case InstalledChimeSounds::Id:
+        return aEncoder.EncodeList([this](const auto & encoder) -> CHIP_ERROR {
+            uint8_t index  = 0;
+            CHIP_ERROR err = CHIP_NO_ERROR;
+
+            // runs through all the Chime sounds available one by one
+            do
+            {
+                Chime::Structs::ChimeSoundStruct::Type chimeSound;
+                if ((err = this->mDelegate.GetChimeSoundByIndex(index, chimeSound)) == CHIP_NO_ERROR)
+                {
+                    ReturnErrorOnFailure(encoder.Encode(chimeSound));
+                    index++;
+                }
+            } while (err == CHIP_NO_ERROR);
+
+            if (err == CHIP_ERROR_NOT_FOUND)
+            {
+                return CHIP_NO_ERROR;
+            }
+            return err;
+        });
+
+    case ActiveChimeID::Id:
+        return aEncoder.Encode(mDelegate.GetActiveChimeId());
```

Comment:
So if something that's not en explicit write changes the active chime id or the enabled state, the delegate is responsible for dealing with the dirty marking?  See also https://github.com/project-chip/connectedhomeip/pull/35729#discussion_r1772149438

Reply:
Moved all the attributes into the server to avoid this, delegate just responsible for the chime details list, and have added comments about that.